packages/node_modules/@ciscospark/widget-message/src/container.js

```diff
@@ -608,22 +598,23 @@ export class MessageWidget extends Component {
     const {
       conversationId,
       firstActivity,
-      sparkInstance,
-      widgetMessage
+      sparkInstance
     } = props;
 
-    props.setScrollPosition({scrollTop: activityList.getScrollTop()});
-
-    if (activityList.isScrolledToBottom() && document.hasFocus()) {
-      props.showScrollToBottomButton(false);
-      props.updateHasNewMessage(false);
-      this.acknowledgeLastActivity();
+    if (activityList.isScrolledToBottom()) {
+      props.setScrolledUp(false);
+      if (document.hasFocus()) {
+        this.acknowledgeLastActivity();
+      }
     }
-    else if (!widgetMessage.get('showScrollToBottomButton')) {
-      props.showScrollToBottomButton(true);
+    else {
+      props.setScrolledUp(true);
     }
 
     if (activityList.isScrolledToTop() && firstActivity.verb !== 'create') {
+      // Store scroll position before loading messages so the window
+      // doesn't jump after they load
+      props.setScrollPosition({scrollTop: activityList.getScrollTop()});
```

Comment:
Scroll position is only used when loading previous activities and was causing weird jumps when scrolling near the bottom (~150px, where the `isScrolledToBottom` buffer comes into play), so I only save scroll position when we've scrolled up and start loading.